File: sdk/logging/dotnet/PrimusSaaS.Logging/Core/LogLevel.cs
namespace PrimusSaaS.Logging.Core;

/// <summary>
/// Log severity levels
/// </summary>
public enum LogLevel
{
    /// <summary>
    /// Detailed diagnostic information
    /// </summary>
    Debug = 0,

    /// <summary>
    /// Informational messages
    /// </summary>
    Info = 1,

    /// <summary>
    /// Warning messages
    /// </summary>
    Warning = 2,

    /// <summary>
    /// Error messages
    /// </summary>
    Error = 3,

    /// <summary>
    /// Critical failures
    /// </summary>
    Critical = 4
}

File: sdk/logging/nodejs/src/core/LogLevel.ts
/**
 * Log levels supported by the logger
 */
export enum LogLevel {
    DEBUG = 'DEBUG',
    INFO = 'INFO',
    WARNING = 'WARNING',
    ERROR = 'ERROR',
    CRITICAL = 'CRITICAL'
}

/**
 * Numeric values for log level comparison
 */
export const LogLevelValue: Record<LogLevel, number> = {
    [LogLevel.DEBUG]: 0,
    [LogLevel.INFO]: 1,
    [LogLevel.WARNING]: 2,
    [LogLevel.ERROR]: 3,
    [LogLevel.CRITICAL]: 4
};
